api/samples/CMakeLists.txt

```diff
@@ -150,13 +150,20 @@ if (OPTIMIZE_SAMPLES)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OPT_CFLAGS}")
 endif ()
 
-function (add_sample_client name source_file_list extension_list)
+function (add_sample_client name source_file_list extension_list show_results)
   add_library(${name} SHARED ${source_file_list})
   configure_DynamoRIO_client(${name})
   foreach (ext ${extension_list})
     use_DynamoRIO_extension(${name} ${ext})
   endforeach (ext)
 
+  # Some samples actually have more tests than just simple sanity tests.
+  # The show_results is a passed flag that indicates that we should override
+  # SHOW_RESULTS_DEFAULT (if is turned off) for this sample.
+  if (NOT SHOW_RESULTS_DEFAULT AND show_results)
```

Comment:
This should look at SHOW_RESULTS, not SHOW_RESULTS_DEFAULT, right?  I.e., at the actual value.  Is the goal of checking that just to avoid a duplicate -DSHOW_RESULTS?

Reply:
Actually, what we want is that we always show results, when building tests. Therefore, I should test BUILD_TESTS AND show_results. I just used SHOW_RESULTS_DEFAULT as a proxy which perhaps caused confusion.